ReactViewControl/ViewModuleContainer.cs

```diff
@@ -51,17 +39,20 @@ public abstract class ViewModuleContainer : IViewModule {
 
         string[] IViewModule.Events => Events;
 
-        string[] IViewModule.DependencyJsSources => DependencyJsSourcesCache.Value;
+        string[] IViewModule.DependencyJsSources => DependenciesProvider.GetJsDependencies(MainJsSource);
 
-        string[] IViewModule.CssSources => CssSourcesCache.Value;
+        string[] IViewModule.CssSources => DependenciesProvider.GetCssDependencies(MainJsSource);
 
         KeyValuePair<string, object>[] IViewModule.PropertiesValues => PropertiesValues;
 
-        void IViewModule.Bind(IFrame frame, IChildViewHost childViewHost) {
+        void IViewModule.Bind(IFrame frame, IChildViewHost childViewHost, IModuleDependenciesProvider moduleDependenciesProvider) {
```

Comment:
why is the bind receiving a moduleDependenciesProvider?

Reply:
my problem is that there's a fallback to a FileDependenciesProvider... it seems that somehow there can be a mix of providers over time

Reply:
This is only set in hot-reload mode, otherwise it will always be null and default to FileDependenciesProvider - same behavior as we have today. I understand your concern, but that will only be true when in hot reload mode.

We need to set it here because we will always start with FileDependenciesProvider during initialization... Otherwise we would have to change generated code, which is not desirable since HotReloadProvider is not known and can be anything else

Reply:
I think it would be better to have a setter... than using the bind method.

Reply:
I mean a factory of providers... not the provider instance